### PR TITLE
Fix Participant import, add tests

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix Participant import, add tests

Currently includes - https://github.com/civicrm/civicrm-core/pull/23692 which I will rebase out

Before
----------------------------------------
Fatal error in MapField

After
----------------------------------------
Error fixed - this also adds  full-flow test & makes the same changes as recently made for Membership.

In the process of this & QA I found errors affecting the other imports too
- campaign options were not being handled
- there was a bug in SaveMappings meaning it would not add new fields to an existing saved mapping
- comma separated values were not being trimmed - I think this was pre-existing in Participant import but 'Attendee, Volunteer' should be handled correctly and now is & is tested

General updates to the way import works - per https://github.com/civicrm/civicrm-core/pull/23689

The underlying changes are

- switch to storing field names not labels in civicrm_mapping_filed
use a persistent table to hold the data being imported
- download errors on the fly from this table rather than pre-generated csvs
- stop using the form get and set methods to pass information around (use the userJob table)
- standardise field validation and transformation based on metadata - when a field is retrieved using getTransformedValue the - - returned value will already have dates, money, option values etc transformed to the right format. If they are invalid they will be transformed to invalid_import_value - this makes most of the remaining formatting and validation pointless

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/Import/Parser/Contact.php [hunk 1]
```diff
@@ -92,34 +92,6 @@ protected function isParseStreetAddress() : bool {
...
   }
 
-  /**
-   * Is this a case where the user has opted to update existing contacts.
-   *
-   * @return bool
-   *
-   * @throws \API_Exception
-   */
-  private function isUpdateExistingContacts(): bool {
```
Comment (eileenmcnaughton) [inline CRM/Contact/Import/Parser/Contact.php:102]: moved to parent

**Comment by eileenmcnaughton** [comment]
@monishdeb are you able look at this one - it's the same as membership

**Comment by aydun** [comment]
Strange help text - "Participants made by ...":
```
 Select 'Individual' if you are importing Participants made by individual persons. Select 'Organization' or 'Household' if you are importing Participants made by contacts of that type. 
```

Got error after hitting Continue on Step 1 (upload data)
```
( ! ) Fatal error: Declaration of CRM_Event_Import_Parser_Participant::validateValues($values) must be compatible with CRM_Import_Parser::validateValues(array $values): void in /home/jenkins/bknix-dfl/build/core-23703-3lxzt/web/sites/all/modules/civicrm/CRM/Event/Import/Parser/Participant.php on line 124
```

File: CRM/Upgrade/Incremental/php/FiveFiftyOne.php [hunk 2]
```diff
@@ -126,6 +126,39 @@ public static function convertMappingFieldLabelsToNames(): bool {
       }
     }
 
+    // Participant fields...
+    // Yes - I know they could be combined - but it's also less confusing this way.
```
Comment (colemanw) [inline CRM/Upgrade/Incremental/php/FiveFiftyOne.php:130]: This comment isn't crystal-clear to me. What could be combined where?

**Comment by colemanw** [comment]
Happy to see this merged as the test coverage is good.

**Comment by eileenmcnaughton** [comment]
there is still some breakage in here - fixing now - but the problem is that managing the PRs is what is breaking things - so now I'm re-fixing stuff  :-( - but I'll have that updated in a few & I think getting it merged will actually take use forwards as it will settle stuff down - will fix the newly broken stuff & then we can re-review once merged

**Comment by eileenmcnaughton** [comment]
Just adding screenshots to show the above is fixed - 

image

image

image

image


And then with the mapping fixed

image


image

**Comment by eileenmcnaughton** [comment]
@aydun can you re-test this once merged? I think that will give us more stability and it is actively broken now without this merged

**Comment by eileenmcnaughton** [comment]
Thanks heaps @aydun - addressing your feedback over at 
https://github.com/civicrm/civicrm-core/pull/23733

**Pull request merged**